### PR TITLE
PUB-1909 View Audit Logs

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1909

### Change description ###

Add filter option to view audit logs function

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AuditService.java [hunk 1]
```diff
@@ -33,8 +37,29 @@ public AuditService(AuditRepository auditRepository) {
      * @param pageable The pageable object to query by.
      * @return Returns the audit logs in a page.
      */
-    public Page<AuditLog> getAllAuditLogs(Pageable pageable) {
-        return auditRepository.findAllByOrderByTimestampDesc(pageable);
+    public Page<AuditLog> getAllAuditLogs(Pageable pageable, String email, String userId,
+        List<AuditAction> auditActions, LocalDateTime filterStartDate, LocalDateTime filterEndDate) {
+
+        // If user provenance id is supplied then find by an exact match
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AuditService.java:43]: We are not filtering by user provenance ID, just user ID

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/database/AuditRepository.java [hunk 2]
```diff
@@ -5,14 +5,22 @@
...
+import java.util.List;
 import java.util.UUID;
 
 public interface AuditRepository extends JpaRepository<AuditLog, UUID> {
 
     @Transactional
     void deleteAllByTimestampBefore(LocalDateTime timestamp);
 
-    Page<AuditLog> findAllByOrderByTimestampDesc(Pageable pageable);
+    Page<AuditLog> findAllByUserEmailLikeIgnoreCaseAndUserIdLikeAndActionInAndTimestampBetweenOrderByTimestampDesc(
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/account/management/database/AuditRepository.java:19]: I am not sure it is better but use the default JPA method provided by spring boot here, or use our owned constructed SQL query. The sticking point here is the timestamp, where we have to pass in an arbitrary start and end time from the front end. If we are using our owned query, we can pass in null and have the query check for null timestamp or a date timestamp by stripping off the time bit from the data. Happy to discuss :)

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AuditService.java (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AuditService.java]: fixed.

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/database/AuditRepository.java (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/database/AuditRepository.java:25]: As discussed this morning in Dev Standup, it's fixed.

**Review by KianKwa — APPROVED** [review]

**Pull request merged**